src/test_util/components.rs

```diff
@@ -164,6 +180,119 @@ impl ComponentTester {
     }
 }
 
+/// Convenience wrapper for running source tests
+pub async fn assert_source_compliance<F, T>(tags: &[&str], f: F) -> T
+where
+    F: Future<Output = T>,
```

Comment:
Any preference for this over `impl Trait` (just saw that discussion in another PR FWIW)?
```suggestion
pub async fn assert_source_compliance<T>(tags: &[&str], f: impl Future<Output = T>) -> T
```